lib/resty/radixtree.lua

```diff
@@ -216,7 +218,7 @@ local function insert_route(self, opts)
     end
 
     local data_idx = radix.radix_tree_find(self.tree, path, #path)
-    if data_idx then
+    if data_idx ~= radix.raxNotFound then
```

Comment:
Instead of leaking the special mark outside, it would be better to handle it inside `radix_tree_find`.

Reply:
sounds reasonable